Automattic/pocket-casts-android
# Proposal: Rename Playable to Episode

I've found our usage of the `Playable` interface to be a bit confusing in a couple of ways.

First, currently the Playable interface is implemented by both `Episode` and `UserEpisode`. I think it's a bit confusing that a `UserEpisode` is not actually an `Episode`.

Second, "episode" feels like a more natural word, and I suspect that is why there are `Playable` variables that are named "episode" in [many places](https://github.com/Automattic/pocket-casts-android/search?q=%22episode%3A+Playable%22) in the project.

For that reason, I thought it might be nice to rename our `Playable` interface to be `Episode`, and to rename the `Episode` to be a `PodcastEpisode` (certainly open to better suggestions for that name). 

This is certainly not addressing any critical issues, it's just a quick change I thought might be a nice quality-of-life improvement for development. So if for any reason you think this isn't worthwhile, just let me know (particularly you, @geekygecko , since you know the history behind `Playable` much better than me). I'm fine closing this. 

> **Note**
> The diff on this PR is a just a little big, but the changes are almost entirely automated refactorings. The one notable exception that I think is worth taking a closer look at is that I migrated the "episodes" table to be named "podcast_episodes".

## Testing Instructions

1. Install a previous version of the app
2. Log into an account with some episodes on the. up next list and some podcast subscriptions.
3. Install a build from this PR
4. Smoke test the app a bit to check for breakages.

## Checklist
